From 52abcc6d4509c69d5fd46cea5526cbe4365a6940 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Fri, 27 Jul 2018 22:50:17 +0800
Subject: [PATCH 4276/5242] MLK-19082 drm/bridge: sec-dsim: wait 'ph_tx_done'
 for long packet send

commit  5b9ba520cfec7a6a42510dfed1aff23745b2290d from
https://source.codeaurora.org/external/imx/linux-imx.git

According to a lot of tests, for long packet send, the packet
payload transfer done interrupt will be triggered no later than
the packet header transfer done interrupt. So, to make sure the
long packet has been send to the peripheral completely, wait
'ph_tx_done' interrupt instead of 'pl_tx_done' for long packet
transfer. Otherwise it may cause subsequent packet transfer
failed sometimes.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 69a5f44025c6ecb2fee16c1650a81198b501f284)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/bridge/sec-dsim.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/bridge/sec-dsim.c b/drivers/gpu/drm/bridge/sec-dsim.c
index 15f908f..ecdc939 100644
--- a/drivers/gpu/drm/bridge/sec-dsim.c
+++ b/drivers/gpu/drm/bridge/sec-dsim.c
@@ -575,7 +575,7 @@ static ssize_t sec_mipi_dsim_host_transfer(struct mipi_dsi_host *host,
 						   packet.header,
 						   use_lpm);
 
-		ret = wait_for_completion_timeout(&dsim->pl_tx_done,
+		ret = wait_for_completion_timeout(&dsim->ph_tx_done,
 						  MIPI_FIFO_TIMEOUT);
 		if (!ret) {
 			dev_err(dsim->dev, "wait payload tx done time out\n");
-- 
1.7.9.5

